zcash_client_sqlite/src/wallet.rs

```diff
@@ -1964,6 +1974,70 @@ pub(crate) fn store_transaction_to_be_sent<P: consensus::Parameters>(
         }
     }
 
+    // Add the transaction to the set to be queried for transaction status. This is only necessary
+    // at present for fully transparent transactions, because any transaction with a shielded
+    // component will be detected via ordinary chain scanning and/or nullifier checking.
+    if !detectable_via_scanning {
+        queue_tx_retrieval(
+            wdb.conn.0,
+            std::iter::once(sent_tx.tx().txid()),
+            TxQueryType::Status,
+            None,
+        )?;
+    }
+
+    Ok(())
+}
+
+pub(crate) fn set_transaction_status(
+    conn: &rusqlite::Transaction,
+    txid: TxId,
+    status: TransactionStatus,
+) -> Result<(), SqliteClientError> {
+    match status {
+        TransactionStatus::TxidNotRecognized | TransactionStatus::NotInMainChain => {
+            // If the transaction is now expired, remove it from the retrieval queue.
+            if let Some(chain_tip) = chain_tip_height(conn)? {
+                conn.execute(
+                    "DELETE FROM tx_retrieval_queue WHERE txid IN (
+                        SELECT txid FROM transactions
+                        WHERE txid = :txid AND expiry_height < :chain_tip_minus_lookahead
+                    )",
+                    named_params![
+                        ":txid": txid.as_ref(),
+                        ":chain_tip_minus_lookahead": u32::from(chain_tip).saturating_sub(VERIFY_LOOKAHEAD)
+                    ],
```

Comment:
Same problem as [below](https://github.com/zcash/librustzcash/pull/1473/files#r1711616150)
```suggestion
                    "DELETE FROM tx_retrieval_queue
                     WHERE query_type = :status_type AND txid IN (
                         SELECT txid FROM transactions
                         WHERE txid = :txid AND expiry_height < :chain_tip_minus_lookahead
                     )",
                    named_params![
                        ":txid": txid.as_ref(),
                        ":chain_tip_minus_lookahead": u32::from(chain_tip).saturating_sub(VERIFY_LOOKAHEAD),
                        ":status_type": TxQueryType::Status.code(),
                    ],
```

Reply:
*sigh* I had _just_ finished responding to your previous comment here when you deleted it, and now I've lost the contents of that response.

The status can be updated to "not recognized" via an enhancement request. This could occur if:
* A transaction for the wallet is detected via scanning.
* A rollback occurs, and the transaction is returned to the mempool.
* The transaction expires or is evicted without being re-mined.

In this situation, it is possible that an enhancement request _may never succeed_. We don't want to keep re-trying indefinitely.

Reply:
> *sigh* I had *just* finished responding to your previous comment here when you deleted it, and now I've lost the contents of that response.

Sorry about that. I hadn't thought that deleting a comment deleted replies.

I don't understand your response. The suggestion adds "query_type = :status_type AND" to the query. This is clearly correct, because we only satisfied the Status request, not the Enhancement request. (We didn't update `raw`; that only happens via `put_tx_data` which is called from `store_decrypted_tx`.) So the question is only whether it's necessary, i.e. whether the situation I'm concerned about (a Status request being upgraded to an Enhancement request) can ever occur.

Reply:
There is a unique constraint on `txid`; there can never be more than one entry for a given transaction in the retrieval queue.

Reply:
The comment added in d96bc110555cee26d405896e8e027e055061e654 explains why this is ok.